.github/workflows/e2e.yml

```diff
@@ -91,12 +90,28 @@ jobs:
 
       # Checkouts
 
-      - name: Checkout (frontend)
+      - name: Checkout (Push)
+        if: github.event_name == 'push'
         uses: actions/checkout@v2
 
-      - name: Set REF in env, removing the `refs/` part
+      - name: Checkout (Pull Request Target)
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          # We are in a pull_request_target event - we need to explicitly checkout the PR's head.
+          # Otherwise we would be running against main branch
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set REF in env, removing the `refs/` part (Push)
+        if: github.event_name == 'push'
         run: echo "MATCHING_BRANCH_REF=$(echo $GITHUB_REF | sed 's|refs/||')" >> $GITHUB_ENV
 
+      - name: Set REF in env, removing the `refs/` part (Pull Request Target)
+        if: github.event_name == 'pull_request_target'
+        run: |
+          BRANCH_REF=${{ github.event.pull_request.head.sha }}
```

Comment:
There is no reason for the SHA to match between API and Frontend?

Reply:
@znarf : Sorry, confusing the `sha` with `ref` here. 🤯 Corrected it here and frond-end.